### PR TITLE
refactor: use async API

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
No

### Motivation / Use-Case

use async API

### Breaking Changes
No

### Additional Info
No

### CODE REVIEW

File: lib/Server.js [hunk 1]
```diff
@@ -405,13 +405,29 @@ class Server {
...
+              if (error) {
+                this.logger.error(error);
+                process.exit(2);
+              }
+            }
+          );
         }
 
-        fakeCert = fs.readFileSync(certificatePath);
+        fakeCert = fs.readFile(certificatePath);
```
Comment (anshumanv) [inline lib/Server.js:430]:
```suggestion
        fakeCert = await fs.readFile(certificatePath);
```

File: lib/Server.js [hunk 2]
```diff
@@ -405,13 +405,34 @@ class Server {
...
+              if (error) {
+                this.logger.error(error);
+              }
+            }
+          );
         }
 
-        fakeCert = fs.readFileSync(certificatePath);
+        fakeCert = await new Promise((resolve, reject) => {
+          fs.readFile(certificatePath, (error, data) => {
```
Comment (alexander-akait) [inline lib/Server.js:429]: maybe we can use `fs/promise`?

Comment (snitin315) [reply lib/Server.js:429]: Yes, that would be better.

File: lib/Server.js [hunk 3]
```diff
@@ -2033,10 +2045,10 @@ class Server {
         const net = require("net");
         const socket = new net.Socket();
 
-        socket.on("error", (error) => {
+        socket.on("error", async (error) => {
```
Comment (alexander-akait) [inline lib/Server.js:2048]: It can be async

**Review by alexander-akait — APPROVED** [review]

**Comment by alexander-akait** [comment]
In future we should avoid sync API in favor async where it possible

**Pull request merged**